test/cmd/apply.sh

```diff
@@ -351,6 +351,7 @@ __EOF__
   # First pass, custom resource fails, but crd apply succeeds.
   output_message=$(! kubectl apply -f hack/testdata/multi-resource-4.yaml 2>&1 "${kube_flags[@]:?}")
   kube::test::if_has_string "${output_message}" 'no matches for kind "Widget" in version "example.com/v1"'
+  kubectl wait --for=jsonpath='{.status.conditions[0].status}'=True crd/widgets.example.com
```

Comment:
Maybe we should check all possible Conditions of CRD, especially Established needs to be True.

apiextensions will only provide Discovery services for the Group when it thinks that the Established Condition is True. The code snippet is as follows:

https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/customresource_discovery_controller.go#L90

Reply:
like: 
```bash
kubectl wait --for=condition=Established=true crd/widgets.example.com          
customresourcedefinition.apiextensions.k8s.io/widgets.example.com condition met
```

Reply:
Should we also wait 30s? I think a couple of seconds should be enough ...

Reply:
It'd be very unfortunate to wait some number of seconds and not for some other condition, if Established isn't enough, what is? waiting for it to show up in apiservices?

Reply:
The sign of the successful registration of the custom gvk in the apiserver is:

Established condition is true
Accepted condition is true

But the Established condition is true, which means that Accepted must be true. Therefore, at present, Established is the sign of the successful registration of the custom gvk in the apiserver.

Sleep mentioned by @apelisse can also alleviate this problem to a great extent, but waiting for the above Established condition to be True is a better solution.

Reply:
> https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/customresource_discovery_controller.go#L90

As @sxllwx  pointed out, waiting for util the CRD's Established to be true seems enough here.

Reply:
> Should we also wait 30s? I think a couple of seconds should be enough ...

Sorry, I was super unclear. I wasn't suggesting we add a `sleep 30`, I was suggesting with put a `--timeout=2s` (instead of the default 30s) to the `kubectl wait` command. The condition will certainly go to `Established` within two seconds or never, I don't think it's worth waiting 28 extra seconds.

Reply:
> problem

em.... I misunderstood. sorry.

Regarding the question of how long to time out, I also agree with `2s`. When the Accepted condition is true and under normal load, the controller can complete the formal establishment of the CRD.

Reply:
Got it.
Add a timeout 2s now.
Thanks.